src/components/InForm/InForm.tsx

```diff
@@ -189,11 +208,128 @@ export const InForm: React.FunctionComponent<any> = (props) => {
   const [isModalOpen, { setTrue: showModal, setFalse: hideModal }] =
     useBoolean(false);
 
+  const [isEditModalOpen, { setTrue: showEditModal, setFalse: hideEditModal }] =
+    useBoolean(false);
+
   function reviewRecord() {
     showModal();
   }
 
-  return (
+  function editRecord() {
+    showEditModal();
+  }
+
+  React.useEffect(() => {
+    let persona: SPPersona[] = [];
+    persona = [{ ...userContext.user }];
+
+    setUser(persona);
+  }, [userContext.user]);
+
+  React.useEffect(() => {
+    const loadRequest = async () => {
+      const res = await requestApi.getItemById(props.ReqId);
+      if (res) {
+        setFormData(res);
+      }
+    };
+
+    loadRequest();
+  }, [props.ReqId, requestApi]);
```

Comment:
OPTIONAL: You can probably simplify this by moving it to a custom hook within your requestApi file. See useChecklistItems